src/reactive-membrane.ts

```diff
@@ -84,45 +82,40 @@ export class ReactiveMembrane {
     }
 
     private getReactiveState(value: any): ReactiveState {
-        const membrane = this;
         const {
             objectGraph,
-            valueMutated,
-            valueObserved,
-        } = membrane;
+        } = this;
         value = unwrap(value);
         let reactiveState = objectGraph.get(value);
         if (reactiveState) {
             return reactiveState;
         }
-
-        reactiveState = ObjectDefineProperties(ObjectCreate(null), {
-            reactive: {
-                get(this: ReactiveState) {
-                    const reactiveHandler = new ReactiveProxyHandler(membrane, value, {
-                        valueMutated,
-                        valueObserved,
-                    });
-                    // caching the reactive proxy after the first time it is accessed
-                    const proxy = new Proxy(createShadowTarget(value), reactiveHandler);
-                    ObjectDefineProperty(this, 'reactive', { value: proxy });
-                    return proxy;
-                },
-                configurable: true,
+        const membrane = this;
```

Comment:
I would expect that you dereference the values from `this` after the branch check for performance reason. You should maybe add a comment since this code construct is not obvious.